wlauto/instrumentation/servo_power_monitors/__init__.py

```diff
@@ -0,0 +1,179 @@
+from wlauto.utils.misc import which
+
+
+class ServoPowerMonitor(Instrument):
+
+    name = 'servo_power'
+    description = """
+    Collects power traces using the Chromium OS Servo Board.
+
+    Servo is a debug board used for Chromium OS test and development. Among other uses, it allows
+    access to the built in power monitors (if present) of a Chrome OS device. More information on
+    Servo board can be found in the link bellow:
+
+    https://www.chromium.org/chromium-os/servo
+
+    In order to use this instrument you need to be a sudoer and you need a chroot environment. More
+    information on the chroot environment can be found on the link bellow:
+
+    https://www.chromium.org/chromium-os/developer-guide
+
+    """
+
+    parameters = [
+        Parameter('power_domains', kind=list_of_strings, default=[],
+                  description="""The names of power domains to be monitored by the instrument using servod."""),
+        Parameter('labels', kind=list_of_strings, default=[],
+                  description="""Meaningful labels for each of the monitored domains."""),
+        Parameter('chroot_path', kind=str, default='',
+                  description="""Path to chroot direcory on the host."""),
+        Parameter('sampling_rate', kind=int, default=10,
+                  description="""Samples per second."""),
+        Parameter('board_name', kind=str, mandatory=True,
+                  description="""The name of the board under test."""),
+    ]
+
+    # When trying to initialize servod, it may take some time until the server is up
+    # Therefore we need to poll to identify when the sever has successfully started
+    # servod_max_tries specifies the maximum number of times we will check to see if the server has started
+    # while servod_delay_between_tries is the sleep time between checks.
+    servod_max_tries = 100
+    servod_delay_between_tries = 0.1
+
+    def initialize(self, context):
+        # pylint: disable=access-member-before-definition
+        self.in_chroot = False if which('dut-control') is None else True
+        self.poller = None
+        domains_string = '_mw '.join(self.power_domains + [''])
+        password = ''
+        if not self.in_chroot:
+            self.logger.info('Instrument {} requires sudo acces on this machine'.format(self.name))
+            self.logger.info('You need to be sudoer to use it.')
+            password = getpass.getpass()
+            check = subprocess.call('echo {} | sudo -S ls > /dev/null'.format(password), shell=True)
+            if check:
+                raise InstrumentError('Given password was either wrong or you are not a sudoer')
+        self.server_session = CrosSdkSession(self.chroot_path, password=password)
+        self.cros_session = CrosSdkSession(self.chroot_path, password=password)
+        password = ''
+        self.server_session.send_command('sudo servod -b {b} -c {b}.xml&'.format(b=self.board_name))
```

Comment:
It would be good if starting servo was optional (a configuration parameter).  Sometimes I already have my servod running for serial and I don't want to kill it for `wa`.